Add --all option to plugins docs and also explain downgrading with an example (#1187)
fermyon/developer: content/spin/v2/managing-plugins.md

$ spin plugins uninstall befunge2wasm
```

## Refreshing the Catalogue

The first time you install a plugin from the catalogue, Spin creates a local cache of the catalogue. It continues to use this local cache for future install, list and upgrade commands; this is similar to OS package managers such as `apt`, and avoids rate limiting on the catalogue. However, this means that in order to see new catalogue entries - new plugins or new versions - you must first update the cache. 

To update your local cache of the catalogue, run `spin plugins update`.

## Upgrading Plugins

> To upgrade a plugin to the latest version, first run `spin plugins update` (to refresh the catalogue). 

Reviewer: ```suggestion
To upgrade a plugin to the latest version, first run `spin plugins update` (to refresh the catalogue), then `spin plugins upgrade`. 
```

$ spin plugins upgrade --all
```

> Note: The above example only installs plugins from the catalogue

The following example shows additional upgrade options. Specifically, how to upgrade using the path to a remote plugin manifest and how to upgrade using the path to a local plugin manifest:

<!-- @selectiveCpy -->

```bash
$ spin plugins upgrade --url https://github.com/fermyon/spin-befunge-sdk/releases/download/v1.7.0/befunge2wasm.json
$ spin plugins upgrade --file ~/dev/spin-befunge-sdk/befunge2wasm.json
```

## Downgrading Plugins

By default, Spin will only _upgrade_ plugins. Pass the `--downgrade` flag and specify the `--version` if you want Spin to roll back to an earlier version. The following abridged example (which doesn't list the full console output for simplicity) lists the versions of plugins, downgrades the `js2wasm` to an older version (`0.6.0`) and then lists the versions again to show the results:

<!-- @nocpy -->

```bash
$ spin plugins update
$ spin plugins list
// --snip--
js2wasm 0.6.0
js2wasm 0.6.1 [installed]
$ spin plugins upgrade js2wasm --downgrade --version 0.6.0
$ spin plugins list
// --snip--
js2wasm 0.6.0 [installed]
js2wasm 0.6.1
```

> After downgrading, that the `[installed]` indicator is aligned with the `0.6.0` version of `js2wasm`, as intended in the example.

Reviewer: ```suggestion
After downgrading, the `[installed]` indicator is aligned with the `0.6.0` version of `js2wasm`, as intended in the example.
```

$ spin plugins uninstall befunge2wasm
```

## Refreshing the Catalogue

The first time you install a plugin from the catalogue, Spin creates a local cache of the catalogue. It continues to use this local cache for future install, list and upgrade commands; this is similar to OS package managers such as `apt`, and avoids rate limiting on the catalogue. However, this means that in order to see new catalogue entries - new plugins or new versions - you must first update the cache. 

To update your local cache of the catalogue, run `spin plugins update`.

## Upgrading Plugins

> To upgrade a plugin to the latest version, first run `spin plugins upgrade` (to refresh the catalogue). 

Reviewer 2: ```suggestion
To upgrade a plugin to the latest version, first run `spin plugins update` (to refresh the catalogue). 
```

The `spin plugins upgrade` command has the same options as the `spin plugins install` command (according to whether the plugin comes from the catalogue, a URL, or a file). For more information, see the plugins section of the [Spin CLI Reference documentation](cli-reference#spin-plugins). 

> The `upgrade` command uses your local cache of the catalogue. This might not include recently added plugins or versions. So always remember to run `spin plugins update` to refresh your local cache of the catalogue before performing the `spin plugins upgrade` command.

The following example shows how to upgrade one plugin at a time (i.e. the `js2wasm` plugin):

<!-- @selectiveCpy -->

```bash
$ spin plugins update
$ spin plugins upgrade js2wasm
```

> The following example shows how to upgrade all installed plugins at once:

Reviewer: This only installs plugins from the registry - do we need to express that explicitly?

Author: Good point, will add that for clarity.

<!-- @selectiveCpy -->

```bash
$ spin plugins update

> $ spin plugins --all

Reviewer: ```suggestion
$ spin plugins upgrade --all
```

```

The following example shows additional upgrade options. Specifically, how to upgrade using the path to a remote plugin manifest and how to upgrade using the path to a local plugin manifest:

<!-- @selectiveCpy -->

```bash
$ spin plugins upgrade --url https://github.com/fermyon/spin-befunge-sdk/releases/download/v1.7.0/befunge2wasm.json
$ spin plugins upgrade --file ~/dev/spin-befunge-sdk/befunge2wasm.json
```

## Downgrading Plugins

By default, Spin will only _upgrade_ plugins. Pass the `--downgrade` flag and specify the `--version` if you want Spin to roll back to an earlier version. The following abridged example (which doesn't list the full console output for simplicity) lists the versions of plugins, downgrades the `js2wasm` to an older version (`0.6.0`) and then lists the versions again to show the results:

<!-- @nocpy -->

```bash
$ spin plugins update
$ spin plugins list

> js2wasm 0.1.0

Reviewer: Consider shortening this list - it overwhelmes the line you actually want to show.

(You could also add blank lines above commands to make them stand out from the console output, but I'm not sure if that will be needed if you truncate the output.)

Author: Thanks, will snip

js2wasm 0.2.0
js2wasm 0.3.0
js2wasm 0.4.0
js2wasm 0.5.0
js2wasm 0.5.1
js2wasm 0.6.0
js2wasm 0.6.1 [installed]
$ spin plugins upgrade js2wasm --downgrade --version 0.6.0
$ spin plugins list
js2wasm 0.1.0
js2wasm 0.2.0
js2wasm 0.3.0
js2wasm 0.4.0
js2wasm 0.5.0
js2wasm 0.5.1
js2wasm 0.6.0 [installed]
js2wasm 0.6.1
```

> After the success of the downgrade process, we can see that the `[installed]` indicator is now aligned with the `0.6.0` version of `js2wasm`, as intended in the example.

Reviewer: Suggest simplifying the sentence structure

```suggestion
After downgrading, that the `[installed]` indicator is aligned with the `0.6.0` version of `js2wasm`, as intended in the example.
```
